### PR TITLE
Disable each segment control when no records pass all other filters

### PR DESCRIPTION
**Pull request by foxtree**
on top of #146 

This PR demonstrates that #146 works (at least for this use-case)

### CODE REVIEW

File: src/redux/selectors/index.js [hunk 1]
```diff
@@ -194,8 +178,94 @@ export const dateRangeForAllRecordsSelector = createSelector(
...
+  (items) => !!items.find(({
+    passesFilters: {
+      type,
+      date,
+      inCollection,
+      // showCollectionOnly,
+      // isHighlighted,
+      showHighlightedOnly,
+    },
+  }) => type && date && showHighlightedOnly && inCollection),
```
Comment (gilhsu) [inline src/redux/selectors/index.js:232]: shouldn't `hasAnyCollectionRecordInScope` ignore `showHighlightedOnly` attribute?

File: src/redux/selectors/index.js [hunk 2]
```diff
@@ -194,8 +178,94 @@ export const dateRangeForAllRecordsSelector = createSelector(
...
+  (items) => !!items.find(({
+    passesFilters: {
+      type,
+      date,
+      // inCollection,
+      showCollectionOnly,
+      isHighlighted,
+      // showHighlightedOnly,
+    },
+  }) => type && date && showCollectionOnly && isHighlighted),
```
Comment (gilhsu) [inline src/redux/selectors/index.js:247]: shouldn't `hasAnyHighlightedRecordInScope` ignore `showCollectionOnly` attribute?

**Review by gilhsu — COMMENTED** [review]
The "Clear Collection" and "Clear Highlighted Records" also need to be disabled when no records are in scope respectively. It might be better to import these new selectors into `CatalogModal` where you can then prop drill into `CollectionSegmentControl` and `MarkedSegmentControl` and use them for the clear buttons.

File: src/redux/selectors/index.js (hunk 1 above)
Comment (foxtree) [reply src/redux/selectors/index.js:232]: it has to pass all filters _other than_ showCollectionOnly
(`showCollectionOnly` also looks at the segment-control state)

File: src/redux/selectors/index.js (hunk 2 above)
Comment (foxtree) [reply src/redux/selectors/index.js:247]: it has to pass all filters other than `showHighlightedOnly`

**Review by gilhsu — APPROVED** [review]

**Comment by foxtree** [comment]
> The "Clear Collection" and "Clear Highlighted Records" also need to be disabled when no records are in scope respectively. It might be better to import these new selectors into `CatalogModal` where you can then prop drill into `CollectionSegmentControl` and `MarkedSegmentControl` and use them for the clear buttons.


I just attempted to do this, but there is some ambiguity... I'll open a separate PR.

**Pull request merged**